File: src/ground.lua
local Ground = {}
Ground.__index = Ground

function Ground.new(game)
	local self = setmetatable({}, Ground)
	self.game = game
	self.images = {}
	self.group = display.newGroup()
	self.x = 0
	self.tileWidth = 0

	local j = 0
	while (j * self.tileWidth < game.WIDTH + self.tileWidth * 2) do
		local image = display.newImage(self.group, 'images/ground.png')
		self.tileWidth = image.width
		image.x = math.floor(j * image.width)
		image.y = math.floor(game.HEIGHT - image.height * 0.5)
		j = j + 1
		self.images[j] = image
	end

	return self
end

function Ground:update(delta)
	self.x = self.x + delta * 0.1 * self.game.speed
	self.group.x = -(math.floor(self.x) % self.tileWidth) - self.tileWidth
end

return Ground

File: src/ground.js
define(['assets'], function (Assets) {
	function Ground(game) {
		PIXI.DisplayObjectContainer.call(this);

		var ground = Assets.ground, sprite, j;

		this.game = game;
		this.x = 0;
		this.tileWidth = ground.width;
		this.height = Assets.ground.height;

		for (j = -1; j * ground.width < game.WIDTH + ground.width * 2; j += 1) {
			sprite = new PIXI.Sprite(ground);
			sprite.position.x = Math.floor(j * ground.width);
			sprite.position.y = Math.floor(game.HEIGHT - ground.height);
			this.addChild(sprite);
		}

		return this;
	}

	Ground.prototype = new PIXI.DisplayObjectContainer();

	Ground.prototype.update = function (delta) {
		this.x += delta * 0.1 * this.game.speed;
		this.position.x = -(Math.floor(this.x) % this.tileWidth) - this.tileWidth;
	};

	return Ground;
});
